src/reducers/map-style-updaters.js

```diff
@@ -368,19 +402,24 @@ export const resetMapConfigMapStyleUpdater = (state) => {
  * @returns {Object} nextState
  * @public
  */
-export const loadCustomMapStyleUpdater = (state, {payload: {icon, style, error}}) => ({
+export const loadCustomMapStyleUpdater = (
+  state,
+  {payload: {icon, style, error}}
+) => ({
   ...state,
   inputStyle: {
     ...state.inputStyle,
     // style json and icon will load asynchronously
-    ...(style ? {
-      id: style.id || generateHashId(),
-      // make a copy of the style object
-      style: JSON.parse(JSON.stringify(style)),
-      label: style.name,
-      // gathering layer group info from style json
-      layerGroups: getLayerGroupsFromStyle(style)
-    } : {}),
+    ...(style
+      ? {
+          id: style.id || generateHashId(),
+          // make a copy of the style object
+          style: JSON.parse(JSON.stringify(style)),
```

Comment:
you are trying to clone an object here. Cloning directly a js object is more performant that using JSON.
https://jsperf.com/deep-copy-vs-json-stringify-json-parse/5

Lodash has several methods to support clone and deep clone

Reply:
done